docs/source/en/advanced/loader.md

```diff
@@ -182,6 +182,86 @@ Note:
 - Same name will be override in loading, for example, if we want to override `ctx.ip` we could define ip in application's `app/extend/context.js` directly.
 - See [framework development](./framework.md) for detail application launch order.
 
+### Life Cycles
+
+The framework has provided you several functions to handle during the whole life cycle:
+
+- `configDidLoad`: When all the config files are loaded.
+- `didLoad`: When all the files are loaded.
+- `willReady`: When all the plug-ins are ready.
+- `didReady`: When all the workers are ready.
+- `serverDidReady`: When the server is ready.
+- `beforeClose`: Before the application is closed.
+
+Here're the definations:
+
+```js
+// app.js or agent.js
+class AppBootHook {
+  constructor(app) {
+  }
+
+  configDidLoad() {
+    // Config, plugin files have loaded.
+  }
+
+  async didLoad() {
+    // All files have loaded, start plugin here.
+  }
+
+  async willReady() {
+    // All plugins have started, can do some thing before app ready
+  }
+
+  async didReady() {
+    // Worker is ready, can do some things
+    // don't need to block the app boot.
+  }
+
+  async serverDidReady() {
+    // Server is listening.
+  }
+
+  async beforeClose() {
+    // Do some thing before app close.
+  }
+}
+
+module.exports = AppBootHook;
+```
+
+The framework will automatically load and initialize this class after developers have defined `app.js` and `agenet.js` in the form of class, and it will call the corresponding methods during each of the life cycles.
+
+Here's the image of starting process:
+
+![](https://user-images.githubusercontent.com/6897780/42219323-24cbd7a8-7efe-11e8-86b9-cf280846aa9f.png)
```

Comment:
Can you replace it with https://user-images.githubusercontent.com/6897780/45463932-bb1f9800-b740-11e8-9f0c-d9649ae8abc0.png?

Reply:
ok, all is done. @killagu